array.c

```diff
@@ -7096,56 +7096,37 @@ rb_ary_combination_size(VALUE ary, VALUE args, VALUE eobj)
 
 /*
  *  call-seq:
- *    array.combination(n) {|element| ... } -> self
- *    array.combination(n) -> new_enumerator
+ *    combination(n) {|element| ... } -> self
+ *    combination(n) -> new_enumerator
  *
- *  Calls the block, if given, with combinations of elements of +self+;
- *  returns +self+. The order of combinations is indeterminate.
- *
- *  When a block and an in-range positive Integer argument +n+ (<tt>0 < n <= self.size</tt>)
- *  are given, calls the block with all +n+-tuple combinations of +self+.
- *
- *  Example:
+ *  With a block given and value +n+ is an
+ *  {integer-convertible object}[rdoc-ref:implicit_conversion.rdoc@Integer-Convertible+Objects]
+ *  in range <tt>1..self.size</tt>,
+ *  calls the block with each +n+-tuple combination of the elements of +self+;
+ *  returns +self+:
  *
- *    a = [0, 1, 2]
- *    a.combination(2) {|combination| p combination }
+ *    a = %w[a b c]                                   # => ["a", "b", "c"]
+ *    a.combination(2) {|combination| p combination } # => ["a", "b", "c"]
  *
  *  Output:
  *
- *    [0, 1]
- *    [0, 2]
- *    [1, 2]
- *
- *  Another example:
+ *    ["a", "b"]
+ *    ["a", "c"]
+ *    ["b", "c"]
  *
- *    a = [0, 1, 2]
- *    a.combination(3) {|combination| p combination }
+ *  The order of the combinations is indeterminate.
  *
- *  Output:
- *
- *    [0, 1, 2]
+ *  When +n+ is zero, calls the block once with a new empty array.
  *
- *  When +n+ is zero, calls the block once with a new empty +Array+:
- *
- *    a = [0, 1, 2]
- *    a1 = a.combination(0) {|combination| p combination }
- *
- *  Output:
- *
- *    []
- *
- *  When +n+ is out of range (negative or larger than <tt>self.size</tt>),
+ *  When +n+ is out-of-range or +self+ is empty,
  *  does not call the block:
```

Comment:
I've made a change that I think makes this correct, but I'm not sure I've stated it in the best way.

Reply:
I think we should keep the original text, especially the  `negative or larger than <tt>self.size</tt>` part, since in this new documentation it's ambiguous what happens when `n` is negative.

Reply:
Modified.  (Fixed?)